cpu/esp32/irq_arch.c

```diff
@@ -83,6 +83,9 @@ static const struct intr_handle_data_t _irq_data_table[] = {
 #if defined(CPU_FAM_ESP32S2) || defined(CPU_FAM_ESP32S3)
     { ETS_USB_INTR_SOURCE, CPU_INUM_USB, 1 },
 #endif
+#if defined(CPU_INUM_SERIAL_JTAG)
+    { ETS_USB_SERIAL_JTAG_INTR_SOURCE, CPU_INUM_SERIAL_JTAG, 1 },
```

Comment:
```suggestion
    { ETS_USB_SERIAL_JTAG_INTR_SOURCE, CPU_INUM_SERIAL_JTAG, 5 },
```
See my comment above.

Reply:
Hm with that, stdio is no longer working.
The highest I can set it to is 3

Reply:
This is wrong again. Either the level is defined to be 5 if `CPU_INUM_SERIAL_JTAG=26` is used or `CPU_INUM_SERIAL_JTAG` is set to 10 and then the level has to be be 1. The mapping of interrupt numbers and priorities is hardwired in Xtensa cores and can't be set by software.

My concern is that level 5 is much too high because it is higher than cache error exception and WDTs. Therefore, I suggested to use interrupt 10 with level 1.

The whole mapping, I found in past in any former ESP-IDF version in `espressif/soc/soc.hext ended by a column with RIOT usage:
```
/**************************************************************************
 *     Int Level   Type           PRO CPU usage        APP CPU uasge        RIOT            Comment
 *      0   1      extern level   WMAC                 Reserved             -               wDev_ProcessFiq
 *      1   1      extern level   BT/BLE Host HCI DMA  BT/BLE Host HCI DMA
 *      2   1      extern level                                             GPIO
 *      3   1      extern level                                             CAN
 *      4   1      extern level   WBB                                       UART
 *      5   1      extern level   BT/BLE Controller    BT/BLE Controller    -
 *      6   1      timer          FreeRTOS Tick(L1)    FreeRTOS Tick(L1)    -               CCOMPARE0
 *      7   1      software       BT/BLE VHCI          BT/BLE VHCI          -
 *      8   1      extern level   BT/BLE BB(RX/TX)     BT/BLE BB(RX/TX)     USB
 *      9   1      extern level                                             RTC
 *      10  1      extern edge
 *      11  3      profiling
 *      12  1      extern level                                             I2C
 *      13  1      extern level                                             WDT
 *      14  7      nmi            Reserved             Reserved
 *      15  3      timer          FreeRTOS Tick(L3)    FreeRTOS Tick(L3)    -               CCOMPARE1
 *      16  5      timer                                                    -               CCOMPARE2
 *      17  1      extern level                                             thread_yield_higher
 *      18  1      extern level                                             ETH
 *      19  2      extern level                                             TIMER
 *      20  2      extern level                                             FRC2 ESP-IDF    FRC2 legacy
 *      21  2      extern level                                             BLE
 *      22  3      extern edge
 *      23  3      extern level                                             
 *      24  4      extern level   TG1_WDT
 *      25  4      extern level   CACHEERR
 *      26  5      extern level
 *      27  3      extern level   Reserved             Reserved
 *      28  4      extern edge    DPORT ACCESS         DPORT ACCESS                         Not in single core
 *      29  3      software       Reserved             Reserved
 *      30  4      extern edge    Reserved             Reserved
 *      31  5      extern level
 **************************************************************************
```

Reply:
hm I think I want a level triggered IRQ for stdout though

Reply:
Does it finally matter?

Reply:
It doesn't make a difference on the ESP32-C3.

But there the level is configurable, so maybe software already does configure it to be a level interrupt?

Reply:
I think it is time to implement the dynamic interrupt allocation as ESP-IDF `intr_alloc` function does where the application just specify requirements like edge/level, priority, ... and the `intr_alloc` function assigns the next free interrupt that satisfies the reuirements.

Reply:
I'm not really sure whether all the BT/BLE interrupt are really used. It would helpful to record a trace which interrupts by the ESP-IDF and the WiFi/BLE libraries are really allocated.